Docs/Examples.md

```diff
@@ -0,0 +1,195 @@
+Examples of O language programs
+===
+
+**note:** unless example is set of methods or classes, it is implies main body:
+
+```java
+class Main is
+    this is
+        // example body
+    end
+```
+
+
+## Valid examples
+
+1. Lists and `Sum()`
+```java
+fibs = []
+a = 1
+b = 1
+while a.Less(10000) loop
+    t = a
+    a = a.Plus(b)
+    b = t
+    fibs.Append(b)
+end
+
+IO().WriteLine(fibs.Sum().ToString())
+```
+
+2. Dicts
+
+```java
+string = "The quick brown fox jumps over the lazy dog"
+count_dct = {}
+i = 0
+
+// construct count dict
+while i.Less(string.Length()) loop
+    key = string.At(i)
+    if count_dct.Exists(key).Not() then
+        count_dct.Set(key, 0)
+    end
+    count_dct.Set(key, count_dct.Get(key).Plus(1))
+end
+
+// print dict
+IO().Write("{")
+var keys = dict.Keys()
+var i = 0
+while i.Less(keys.Length()) loop
+    var key = keys.Get(i)
+    IO().Write(key)
+    IO().Write(": ")
+    IO().Write(dict.Get(key).ToString())
+    IO().Write(", ")
+    i = i.Plus(1)
+end
+IO().Write("}")
+```
+
+
+
+## Invalid examples (compile time errors)
+
+```java
+lst = [] // <-- error: cannot infer type of `lst`, provide type hint
+a = lst.Get(0)
+```
+
+
+```java
+fibs = []
+fibs.Append(1)
+fibs.Append("2") // <- error: List<Integer>.Append 
+                 // expects first agrument of type Integer
+```
+
+```java
+fibs: List<Integer> = []
+fibs.Append("2") // <- error: List<Integer>.Append 
+                 // expects first agrument of type Integer
+```
+
+```java
+lst = [1, False] // <-- error: cannot construct lists of different type
```

Comment:
`cannot construct lists of different type` -> `cannot construct a list from elements of different types`